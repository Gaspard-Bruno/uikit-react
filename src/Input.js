import cx from 'classnames'
import { PropTypes } from 'react'

const Input = ({
  danger,
  success,
  blank,
  large,
  small,
  className: customClassName,
  ...other,
}) => {
  const className = cx(customClassName, {
    'uk-form-danger': danger,
    'uk-form-success': success,
    'uk-form-blank': blank,
    'uk-form-large': large,
    'uk-form-small': small,
  })
  return (
    <input {...other} className={className} />
  )
}

Input.propTypes = {
  danger: PropTypes.bool,
  success: PropTypes.bool,
}

export default Input
