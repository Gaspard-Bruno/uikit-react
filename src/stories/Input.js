import { storiesOf } from '@kadira/storybook'
import { Input } from 'uikit-react'

// This is to work around: https://github.com/kadirahq/react-storybook-addon-info/issues/26#issuecomment-229029177
Input.displayName = 'Input'

storiesOf('Input', module)
  .addWithInfo('Basic Usage', '', () => (
    <form className="uk-margin-bottom uk-form">
      <Input placeholder="Default" /> &nbsp;
      <Input danger placeholder="Danger" /> &nbsp;
      <Input success placeholder="Success" />
    </form>
  ), { header: false, inline: true, propTables: [Input] })
