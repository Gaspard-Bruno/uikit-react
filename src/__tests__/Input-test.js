import renderer from 'react-test-renderer'

import Input from '../Input'

describe('Input', () => {
  it('renders correctly', () => {
    expect(renderer.create(
      <Input />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input className="uk-search-field" type="search" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input danger />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input success />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input blank />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input large />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input small />
    ).toJSON()).toMatchSnapshot()
  })
})
