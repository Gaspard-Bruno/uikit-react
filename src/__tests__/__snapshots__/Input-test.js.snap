exports[`Input renders correctly 1`] = `
<input
  className="" />
`;

exports[`Input renders correctly 2`] = `
<input
  className="uk-search-field"
  type="search" />
`;

exports[`Input renders correctly 3`] = `
<input
  className="uk-form-danger" />
`;

exports[`Input renders correctly 4`] = `
<input
  className="uk-form-success" />
`;

exports[`Input renders correctly 5`] = `
<input
  className="uk-form-blank" />
`;

exports[`Input renders correctly 6`] = `
<input
  className="uk-form-large" />
`;

exports[`Input renders correctly 7`] = `
<input
  className="uk-form-small" />
`;
